zkevm-circuits/src/mpt_circuit/account_leaf.rs

```diff
@@ -261,6 +271,17 @@ impl<F: Field> AccountLeafConfig<F> {
                     storage_items[is_s.idx()].word(),
                 );
             }
+
+            ifx! {or::expr(&[config.is_mod_extension[0].clone(), config.is_mod_extension[1].clone()]) => {
+                config.mod_extension = ModExtensionGadget::configure(
+                    meta,
+                    cb,
+                    ctx.clone(),
+                    parent_data,
+                    key_data,
+                );
+            }};
```

Comment:
As expected the block is executed when the condition is true.

For passing the condition to the gadget - that would mean having one big if statement in `ModExtensionGadget` or having the condition in each constraint statement. I think the current code is somehow easier to read?

Reply:
Any specific concerns here about the use of `ifx!` @ChihChengLiang ? I think `ifx` is being used here similarly to other places, so not sure how the specific concerns here.

Some of the gadgets are created inside the `ifx!` because otherwise things also get quite a bit messier to keep track of all the different paths and inputs, but regardless of the condition they are always created.

Reply:
Hi @Brechtpd @miha-stopar,
I've managed to convince myself that the current way of `ifx!` works with this mini example https://github.com/privacy-scaling-explorations/zkevm-circuits/commit/1dd12cadca0094884010724af12dffb8f2887ba7. The concern is resolved. Also thank you all for explaining to me.